graspologic/plot/plot.py

```diff
@@ -673,6 +677,253 @@ def pairplot(
+    # Handle X
+    if not isinstance(X, (list, np.ndarray)):
+        msg = "X must be array-like, not {}.".format(type(X))
+        raise TypeError(msg)
+
+    # Handle labels
+    if labels is not None:
+        if not isinstance(labels, (list, np.ndarray)):
+            msg = "Y must be array-like or list, not {}.".format(type(labels))
+            raise TypeError(msg)
+        elif X.shape[0] != len(labels):
+            msg = "Expected length {}, but got length {} instead for Y.".format(
+                X.shape[0], len(labels)
+            )
+            raise ValueError(msg)
+    # Handle gmm
+    if gmm is None:
+        msg = "You must input a sklearn.mixture.GaussianMixture"
+        raise TypeError(msg)
+    # get relevant features from the gmm model
+    Y_, means, covariances = gmm.predict(X), gmm.means_, gmm.covariances_
+    # plot only a scatter plot with only two feature columns
+    lab_dict, ind_to_lab = {}, {}
+    if labels:
+        set_labs = sorted(set(labels))
+        counter = 0
+        for lab in set_labs:
+            lab_dict[lab] = counter
+            ind_to_lab[counter] = lab
+            counter += 1
+        Y_ = np.asarray([lab_dict[labels[i]] for i in range(len(labels))])
+    colors = dict(zip(np.unique(Y_), colors))
+    with sns.plotting_context(context=context, font_scale=font_scale):
+        if X.shape[1] == 2:
+            pairplot, axes = plt.subplots(1, 1, figsize=(12, 12))
+            plot_ellipse(
+                X,
+                Y_,
+                means,
+                covariances,
+                axes,
+                0,
+                1,
+                colors,
+                lab_dict,
+                ind_to_lab,
+                palette=palette,
+            )
+            axes.set_ylabel("Dim " + str(0))
+            axes.set_xlabel("Dim " + str(1))
+            axes.label_outer()
+            axes.spines["right"].set_visible(False)
+            axes.spines["top"].set_visible(False)
+            if title:
+                pairplot.suptitle(title)
+            else:
+                pairplot.suptitle(
+                    "Fit a Gaussian mixture using "
+                    + str(means.shape[0])
+                    + " components"
+                )
+            plt.subplots_adjust(right=0.8)
+            return pairplot
+        # for more than two features compared
+        pairplot, axes = plt.subplots(X.shape[1], X.shape[1], figsize=(12, 12))
+        for k in range(X.shape[1]):
+            for j in range(X.shape[1]):
+                if k == j:
+                    for t in range(X.shape[1]):
+                        sns.distplot(
+                            X[Y_ == t, k], kde=True, ax=axes[k, k], color=colors[t]
+                        )
+                    axes[k, j].set_xticks([]), axes[k, j].set_yticks([])
+                    axes[k, j].set_ylabel(None)
+                    axes[k, j].set_ylabel(j), axes[k, j].set_xlabel(k)
```

Comment:
seems like all this could be done a lot more simply with seaborn's `PairGrid`
https://seaborn.pydata.org/generated/seaborn.PairGrid.html

Reply:
I thought about using pairgrid but i enjoy having the control that i had with using matplotlib. While seaborn is nice it is just a little annoying to manipulate in my opinion.